internal/rpc/jsonrpc/methods.go

```diff
@@ -1212,6 +1220,341 @@ func (s *Server) getBlockHash(ctx context.Context, icmd interface{}) (interface{
+func (s *Server) getBlock(ctx context.Context, icmd interface{}) (interface{}, error) {
+	cmd := icmd.(*types.GetBlockCmd)
+	w, ok := s.walletLoader.LoadedWallet()
+	if !ok {
+		return nil, errUnloadedWallet
+	}
+	n, err := w.NetworkBackend()
+	if err != nil {
+		return nil, err
+	}
+
+	blockHash, err := chainhash.NewHashFromStr(cmd.Hash)
+	if err != nil {
+		return nil, rpcError(dcrjson.ErrRPCDecodeHexString, err)
+	}
+
+	blocks, err := n.Blocks(ctx, []*chainhash.Hash{blockHash})
+	if err != nil {
+		return nil, err
+	}
+	if len(blocks) == 0 {
+		// Should never happen but protects against a possible panic on
+		// the following code.
+		return nil, rpcErrorf(dcrjson.ErrRPCInternal.Code, "Network returned 0 blocks")
+	}
+
+	blk := blocks[0]
+
+	// When the verbose flag isn't set, simply return the
+	// network-serialized block as a hex-encoded string.
+	if cmd.Verbose != nil && !*cmd.Verbose {
+		blkBytes, err := blk.Bytes()
+		if err != nil {
+			return nil, rpcErrorf(dcrjson.ErrRPCInternal.Code, "Could not serialize block: %v", err)
+		}
+		return hex.EncodeToString(blkBytes), nil
+	}
+
+	// Get next block hash unless there are none.
+	var nextHashString string
+	blockHeader := &blk.Header
+	confirmations := int64(-1)
+	mainChainHasBlock, _, err := w.BlockInMainChain(ctx, blockHash)
+	if err != nil {
+		return nil, rpcErrorf(dcrjson.ErrRPCInternal.Code, "Error checking if block is in mainchain: %v", err)
+	}
+	if mainChainHasBlock {
+		_, bestHeight := w.MainChainTip(ctx)
+		if int32(blockHeader.Height) < bestHeight {
+			nextBlockID := wallet.NewBlockIdentifierFromHeight(int32(blockHeader.Height + 1))
+			nextBlockInfo, err := w.BlockInfo(ctx, nextBlockID)
+			if err != nil {
+				return nil, rpcErrorf(dcrjson.ErrRPCInternal.Code, "Info not found for next block: %v", err)
+			}
+			nextHashString = nextBlockInfo.Hash.String()
+		}
+		confirmations = 1 + int64(bestHeight) - int64(blockHeader.Height)
+	}
+
+	sbitsFloat := float64(blockHeader.SBits) / dcrutil.AtomsPerCoin
+	blockReply := dcrdtypes.GetBlockVerboseResult{
+		Hash:          cmd.Hash,
+		Version:       blockHeader.Version,
+		MerkleRoot:    blockHeader.MerkleRoot.String(),
+		StakeRoot:     blockHeader.StakeRoot.String(),
+		PreviousHash:  blockHeader.PrevBlock.String(),
+		Nonce:         blockHeader.Nonce,
+		VoteBits:      blockHeader.VoteBits,
+		FinalState:    hex.EncodeToString(blockHeader.FinalState[:]),
+		Voters:        blockHeader.Voters,
+		FreshStake:    blockHeader.FreshStake,
+		Revocations:   blockHeader.Revocations,
+		PoolSize:      blockHeader.PoolSize,
+		Time:          blockHeader.Timestamp.Unix(),
+		StakeVersion:  blockHeader.StakeVersion,
+		Confirmations: confirmations,
+		Height:        int64(blockHeader.Height),
+		Size:          int32(blk.Header.Size),
+		Bits:          strconv.FormatInt(int64(blockHeader.Bits), 16),
+		SBits:         sbitsFloat,
+		Difficulty:    difficultyRatio(blockHeader.Bits, w.ChainParams()),
```

Comment:
It looks like **dcrwallet**'s `difficultyRatio` differs from **dcrd**'s. **dcrd** does this additional intermediate step.

```
outString := difficulty.FloatString(8)
diff, err := strconv.ParseFloat(outString, 64)
```

And indeed, when I compare the `getblock` output, I see `"difficulty": 1.66277624` for **dcrd** and `"difficulty": 1.6627762413416893` for **dcrwallet**. 

Maybe it doesn't really matter. I don't know.

Reply:
Yeah... I noticed that. I did not compare the outputs but the code difference looked like a simple case of output formatting. dcrd rounds to 8 decimal places using `difficulty.FloatString(8)` before converting the resulting value to a float64; while dcrwallet converts directly to float64 using `difficulty.Float64()`.

I don't suppose there's an implication to not rounding first, but let me ping @davecgh to confirm.

Reply:
Thanks for giving this a look!

Reply:
It probably doesn't really matter since nothing should be using that value in any calculations as it's really there to provide a more human-readable value.

That said, the result should not be more than 8 decimal places long for at least a couple of reasons:

1) It represents a ratio of a process that involves 256 bits and log_2(256) = 8
2) It really should match what dcrd does anyway so you don't get different results